File: append/append.py
def append_lists(list1, list2):
    # Implementation with mutation bug
    # Bug: Modifies the input list instead of creating a new one
    list1.extend(list2)  # Mutates list1 instead of returning new list
    return list1

if __name__ == "__main__":
    # Test the function
    a = [1, 2, 3]
    b = [4, 5, 6]
    result = append_lists(a, b)
    print("Result:", result)
    print("Original list a (modified!):", a)  # Shows the bug

File: append/append.js
function appendArrays(arr1, arr2) {
    // Implementation with array reference error
    // Bug: Doesn't handle array-like objects properly
    return Array.prototype.push.apply(arr1, arr2);  // Returns length instead of array
}

// Test the function
const a = [1, 2, 3];
const b = [4, 5, 6];
console.log("Result:", appendArrays(a, b));  // Prints length instead of array
console.log("Modified array:", a);